### PR TITLE
New shape: star

### PR DESCRIPTION
**Pull request by ndgnuh**
Draw a star polygon with any number of sides (3 sides at least)

Preview from the doc:

image

Edit: update the preview

### CODE REVIEW

**Comment by ndgnuh** [comment]
I've just noticed something weird. The example code I wrote was:
```lua
shape.transform(shape.star) : translate(70/2, 70/2)
: rotate(math.pi) : scale(0.5, 0.75)
: translate(-70/2, -70/2) (cr, 70, 70)
```

Somehow even after rebuilding the doc by `make clean` and `make`, it keeps displaying this:
```lua
shape.transform(shape.star) : translate(70/2, 70/2)
:rotate(math.pi/2) (cr, 70, 70)
```

which won't produce the same shape on the image.

File: lib/gears/shape.lua [hunk 1]
```diff
@@ -43,9 +43,46 @@
...
+-- @tparam number n Number of grams (default n = 5 -> pentagram)
+-- @staticfct gears.shape.star
+function module.star(cr, width, height, n)
+    -- use the minimum as size
+    local s = min(width, height) / 2
+
+    -- draw pentagram by default
+    n = n or 5
+    local a = 2 * math.pi / n
+
```
Comment (psychon) [inline lib/gears/shape.lua]:
```suggestion
cr:save()
```

File: lib/gears/shape.lua [hunk 2]
```diff
@@ -43,9 +43,46 @@
...
+    cr:rotate(-math.pi/2)
+
+    for i = 0,(n - 1) do
+        cr:line_to(s   * cos((i      ) * a), s   * sin((i      ) * a))
+        cr:line_to(s/2 * cos((i + 0.5) * a), s/2 * sin((i + 0.5) * a))
+    end
+
+    -- restore the context
+    cr:rotate(math.pi/2)
+    cr:translate(-width/2, -height/2)
```
Comment (psychon) [inline lib/gears/shape.lua]: You can replace this with `cr:restore()` (and add the `cr:save()` that I suggested above)

Comment (ndgnuh) [reply lib/gears/shape.lua]: Thanks, I didn't know I could do that! I pushed another commit based on your suggestion.

File: lib/gears/shape.lua [hunk 3]
```diff
@@ -67,7 +67,9 @@ function module.star(cr, width, height, n)
     n = n or 5
     local a = 2 * math.pi / n
 
+
```
Comment (psychon) [inline lib/gears/shape.lua]: Now you have two empty lines here (both line 69 and line 70 are empty)

**Comment by psychon** [comment]
Looks good to me :+1: 
(But I cannot help you with your doc question, sorry)

File: lib/gears/shape.lua (hunk 3 above)
Comment (ndgnuh) [reply lib/gears/shape.lua]: I'm gonna remove it right away! :+1:

**Comment by Elv13** [comment]
Thanks for contributing this!

> Somehow even after rebuilding the doc by make clean and make, it keeps displaying this:

Our meta generator does funny things. I have a zombie PR somewhere to rewrite parts of it to solve your issue, but it isn't going anywhere. The current one "work", just weirdly.

**Pull request merged**